### ISSUE TITLE
go-flags does not work on Google App Engine due to unsafe package

### ISSUE THREAD

**Issue by mpalomas** [jessevdk/go-flags#218]
Hello,

Google App Engine has some restrictions on the Go code we can run, and for instance, the unsafe package is forbidden. Some of the restrictions are listed here: https://cloud.google.com/appengine/docs/go/runtime

You are using the unsafe package (probably for very good reasons) and it prevents us from deploying on App Engine. There may be similar issues with the syscall package according the the Google documentation.

Is there any way to replace the unsafe/pointer usage in your code by something else that would provide similar functionalities? I'm asking naively since I've not looked deeply into your code.

If we need to fork, any recommendation?

Thanks!

**Comment by jessevdk** [comment]
I looked a bit, and the only real need for unsafe is for the syscall that determines the terminal size. This is a really optional feature though. I've created a PR that removes the unecessary use of unsafe, and removes the syscall when the `appengine` build constraint is set.

Could you try the https://github.com/jessevdk/go-flags/pull/220 PR (branch `#218/no-unsafe`) and see if that resolves your issue?

**Comment by mpalomas** [comment]
Yes,  that does indeed solve the problem and I actually ended writing something similar on my dev environment.

But I've also realised something important end of last week (I'm a newbie on App Engine):

Actually, when you write an app against App Engine, your main.go is not an executable, it's a package, a lib (no main function, but init instead). Meaning it just does not makes sense to parse command lines on App Engine... 
So, my final solution is that I have 2 different main.go using conditional compilation, and the for for App Engine simply does not use go-flags at all...

I still wonder whether your PR could be valuable for other use cases in similar restricted runtime environment though.
